src/role_carry.js

```diff
@@ -143,18 +143,20 @@ roles.carry.preMove = function(creep, directions) {
           return true;
         }
         reverse = creep.carry.energy - transferred.transferred > 0;
-      } else if (creep.memory.routing.pathPos === 0 && !(creep.room.storage && creep.room.storage.my && creep.room.storage.isActive())) {
+      } else if (creep.memory.routing.pathPos === 0) {
         creep.drop(RESOURCE_ENERGY);
```

Comment:
So will not creep drop energy anytime ? Are you sure the creep will succesfully transfer after that ? Should be placed l.151 i supose.

Reply:
I have applied these changes to my scripts in the official server for several days.  Carry will drop energy only when `pathPos === 0 && inBase`.  I placed `drop` here because we shouldn't call harvester if we have storage.

Reply:
yeah but we should also not drop if we have storage not full

Reply:
The check is included in `transferToStructures`.  If `transferred` is `false`, carry can't transfer to any structures including storage.

Reply:
alright, missed it